src/start.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getEnvBoolean } from './utils';
+import { startMetrics, MetricsOptions } from './metrics';
+import { startProfiling, ProfilingOptions } from './profiling';
+import { startTracing, TracingOptions } from './tracing';
+
+interface GeneralOptions {
```

Comment:
Metrics and profiling options have a `resource` field, what do you think about adding it to tracing as well? So we can add a common `resource` to `GeneralOptions`

Reply:
So the user could create the Resource themselves before initializing the SDK?
That's the idea in the end. I didn't want to rewrite everything at once. I'll address thsi somehow in another PR, but since the current implementation doesn't take anything away, I'll leave that as-is for now.

Reply:
Note: Metrics does not have a resource field.